### PR TITLE
Add meson build system

### LINKED ISSUE #12: Use Meson as build system
It would be more comfortable compile and cross-compile `CRC-manipulator` using `Meson` as build system.

### PR DESCRIPTION
**Pull request by Luni-4**
Fix #12 

TODO

- [x] Fix cross-compilation for Windows

- [x] Download `catch.hh` for tests only.

### CODE REVIEW

File: lib/meson.build [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    'file.cc',
+    'progress.cc',
+    'util.cc'
+)
+
+crcmanip_lib = library('crcmanip-lib',
+                       lib_src,
+                       config_h,
+                       install: true,
+                       include_directories: incs)
```
Comment (rr-) [inline lib/meson.build]: Please use indenting style like above

File: gui/meson.build [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    'main_window.cc',
+    'drop_line_edit.cc',
+    processed
+]
+
+crcmanip_gui  = executable('crcmanip-gui',
+                           sources: gui_src,
+                           include_directories: incs,
+                           dependencies: qt5widgets,
+                           link_with: crcmanip_lib)
```
Comment (rr-) [inline gui/meson.build]:
```
exampleexample = example(example,
                         example, example,
                         example)
```
→
```
exampleexample = example(
    example,
    example,
    example,
    example
)
```

(everywhere)

**Comment by Luni-4** [comment]
@rr- 

This PR is complete. The binaries can be used as standalone programs on `Windows` because the `qt5` libraries are static. I've also removed the old build system.

Thanks in advance for your review! :)

**Comment by rr-** [comment]
Fails to build on linux mint

```
rr-@lightning:~/src/priv/crc-manipulator$ meson build --buildtype release
The Meson build system
Version: 0.45.1
Source dir: /home/rr-/src/priv/crc-manipulator
Build dir: /home/rr-/src/priv/crc-manipulator/build
Build type: native build

meson.build:1:0: ERROR: Meson version is 0.45.1 but project requires >= 0.49.0.

A full log can be found at /home/rr-/src/priv/crc-manipulator/build/meson-logs/meson-log.txt
```

**Comment by Luni-4** [comment]
> Fails to build on linux mint
> 
> ```
> rr-@lightning:~/src/priv/crc-manipulator$ meson build --buildtype release
> The Meson build system
> Version: 0.45.1
> Source dir: /home/rr-/src/priv/crc-manipulator
> Build dir: /home/rr-/src/priv/crc-manipulator/build
> Build type: native build
> 
> meson.build:1:0: ERROR: Meson version is 0.45.1 but project requires >= 0.49.0.
> 
> A full log can be found at /home/rr-/src/priv/crc-manipulator/build/meson-logs/meson-log.txt
> ```

Please update your meson version to >=0.49.0

**Comment by rr-** [comment]
Sorry, but no. This is what Mint has. It's a big distribution. Do we really need the newest release here?

**Comment by Luni-4** [comment]
> Sorry, but no. This is what Mint has. It's a big distribution. Do we really need the newest release here?
The 0.49 introduced some features that I've used for building this project. The actual newest release is the 0.52. 

You can update it using `Python`, there is no need to change anything else
  
    pip3 install meson ninja

Perhaps should I specify that in the `README`?

**Comment by rr-** [comment]
>The 0.49 introduced some features that I've used for building this project. The actual newest release is the 0.52.
>
>You can update it using Python, there is no need to change anything else
>
>`pip3 install meson ninja`
>Perhaps should I specify that in the README?

Yes, definitely. I didn't know it's Python-only. Thanks!

**Comment by Luni-4** [comment]
> > The 0.49 introduced some features that I've used for building this project. The actual newest release is the 0.52.
> > You can update it using Python, there is no need to change anything else
> > `pip3 install meson ninja`
> > Perhaps should I specify that in the README?
> 
> Yes, definitely. I didn't know it's Python-only. Thanks!

You're welcome! :)
So now does it work?

**Comment by rr-** [comment]
Unfortunately not. Could you please make QT optional?

**Comment by Luni-4** [comment]
> Unfortunately not. Could you please make QT optional?

What's the error? 

So do you want to make the `crcmanip-gui` optional? Yep, it would be possible

**Comment by rr-** [comment]
It can't find qt5 even though I obviously have it in my system (probably some dev packages). Yes, GUI definitely should be optional, precisely because of these issues.

**Comment by Luni-4** [comment]
> It can't find qt5 even though I obviously have it in my system (probably some dev packages). Yes, GUI definitely should be optional, precisely because of these issues.

That's pretty weird. It should find them if they are in the default paths. If you moved them somewhere else, you need to change your `PKG_CONFIG_PATH` though.

**Comment by Luni-4** [comment]
@rr- 

Now you can compile and cross-compile the project without the `crcmanip-gui`.

**Comment by rr-** [comment]
Thanks @Luni-4 I tested it and it worked. Merged manually and added a small fix on master for running the tests.

**Comment by Luni-4** [comment]
@rr- 

Oh, good catch for the wrong variable and formatting, thanks! :)

**Pull request closed without merging**